### DISCUSSION TITLE
Send BEEP command to barcode scanner connected by serial port using Node's SerialPort library

### DISCUSSION THREAD

**Discussion by dpulit** [serialport/node-serialport#2519] [Q&A]
I have a programming problem with my Barcode scanner connected to PC by serial port (COM3).

I'm trying to reading data from the scanner and depending on the specific condition send command to device to beep.

Reading data from scanner is not the problem. The problem is sending command to make a scanner beep.

My code:
```
const { SerialPort } = require('serialport');

var sp = new SerialPort({
    path: 'COM3',
    baudRate: 115200,
});

sp.on('data', function (data) {
    const number = data.toString('utf-8');
    console.log('Data:', number)

    if (number == 5901882110014) {
        sendData();
    }
})

function sendData() {}
```

I want send command to scanner to beep if, for example received barcode is 5901882110014.

How can I approach to this problem?

My scanner is Zebra DS2278: 
https://www.zebra.com/content/dam/zebra_new_ia/en-us/manuals/barcode-scanners/general/ds2278/ds2278-prg-en.pdf 
https://www.zebra.com/content/dam/zebra_new_ia/en-us/manuals/barcode-scanners/software/opos-driver-dg-en.pdf

**Comment by qvrsefv5** [comment]
Hello, quite some time, i am having same issue but have honeywell scanner 
when i try to screen /dev/tty.xxxx 9600  and press ctrl+g the device beeps but i can't programmatically send it the beep did you solve it ? 
ctrl+g should be bel
so my implementation looks likes this 
```const BEL = Buffer.from([0x07]);
      port.write(BEL, (err) => {
        if (err) {
          console.error("Error writing BEL:", err);
        } else {
          console.log("BEL sent, scanner should beep if configured.");
        }
      });```

**Reply by qvrsefv5** [reply]
it works now!
